language/tools/move-cli/src/main.rs

```diff
@@ -189,6 +212,25 @@ impl Move {
     fn get_library_modules(&self) -> Result<Vec<CompiledModule>> {
         self.mode.compiled_modules(&self.get_package_dir())
     }
+
+    /// Preload the storage with library modules (if such modules are not available in the storage)
+    fn preload_storage(&self) -> Result<()> {
+        let storage_dir = maybe_create_dir(&self.storage_dir)?;
+        let view = OnDiskStateView::create(storage_dir.to_path_buf(), &[])?;
```

Comment:
I worry that the `prepare_mode`/`preload_storage` flow is getting a bit too imperative--seems easy to set it up on one control-flow path but not another, or to accidentally set it up multiple times.

Maybe we could make things a bit more functional by changing `OnDiskStateView` to take a `mode` + do what `preload_storage` is doing here (or something else)? We can also address this in a follow-up if  it's too big for here.

Reply:
Yes, this is a bit concerning. Shall I suggest to do it in a follow-up PR as this will be addressing a separate issue.

Reply:
Sounds good!